app/controllers/renewal/WhatYouNeedController.scala

```diff
@@ -15,29 +15,41 @@
  */
 
 package controllers.renewal
-
+import cats.data.OptionT
+import cats.implicits._
+import connectors.DataCacheConnector
 import controllers.{AmlsBaseController, CommonPlayDependencies}
 import javax.inject.{Inject, Singleton}
+import models.businessmatching.{BusinessActivities, BusinessMatching, BusinessMatchingMsbServices}
 import models.registrationprogress.{NotStarted, Section, Started}
-import play.api.mvc.MessagesControllerComponents
+import play.api.mvc.{MessagesControllerComponents, Request}
 import services.RenewalService
 import utils.AuthAction
 import views.html.renewal._
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
-
 @Singleton
-class WhatYouNeedController @Inject()(val authAction: AuthAction,
-                                      val ds: CommonPlayDependencies,
-                                      renewalService: RenewalService,
-                                      val cc: MessagesControllerComponents)(implicit ec: ExecutionContext) extends AmlsBaseController(ds, cc) {
-
+class WhatYouNeedController @Inject()(
+                                       val dataCacheConnector: DataCacheConnector,
+                                       val authAction: AuthAction,
+                                       val ds: CommonPlayDependencies,
+                                       renewalService: RenewalService,
+                                       val cc: MessagesControllerComponents)(implicit ec: ExecutionContext) extends AmlsBaseController(ds, cc) {
   def get = authAction.async {
     implicit request =>
-      renewalService.getSection(request.credId) map {
-        case Section(_,NotStarted | Started,_,_) => Ok(what_you_need())
-        case _ => Redirect(controllers.routes.RegistrationProgressController.get())
-      }
+        (for {
+          cache <- OptionT(dataCacheConnector.fetch[BusinessMatching](request.credId, BusinessMatching.key))
```

Comment:
Can we please rename this to `BusinessMatching` or `bm` as this is only one section of the cache?

Reply:
Done